optaplanner-examples/src/test/java/org/optaplanner/examples/vehiclerouting/app/VehicleRoutingMultiThreadedReproducibilityTest.java

```diff
@@ -0,0 +1,78 @@
+import org.optaplanner.examples.common.app.AbstractTurtleTest;
+import org.optaplanner.examples.common.app.CommonApp;
+import org.optaplanner.examples.vehiclerouting.domain.VehicleRoutingSolution;
+import org.optaplanner.examples.vehiclerouting.persistence.VehicleRoutingImporter;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * The idea is to verify one of the basic requirements of Multithreaded Solving - the reproducibility of results. After
+ * a constant number of steps, every iteration must finish with the same score.
+ */
+public class VehicleRoutingMultiThreadedReproducibilityTest extends AbstractTurtleTest {
+
+    private static final int REPETITION_COUNT = 10;
+
+    private static final int STEP_LIMIT = 5000;
+
+    private static final String MOVE_THREAD_COUNT = "4";
+
+    private static final String DATA_SET = "import/belgium/basic/air/belgium-n50-k10.vrp";
+
+    private final VehicleRoutingApp vehicleRoutingApp = new VehicleRoutingApp();
+
+    private VehicleRoutingSolution[] vehicleRoutingSolutions = new VehicleRoutingSolution[REPETITION_COUNT];
+
+    private SolverFactory<VehicleRoutingSolution> solverFactory;
+
+    @Before
+    public void createUninitializedSolutions() {
+        final VehicleRoutingImporter importer = new VehicleRoutingImporter();
+        for (int i = 0; i < REPETITION_COUNT; i++) {
+            File dataSetFile = new File(CommonApp.determineDataDir(vehicleRoutingApp.getDataDirName()), DATA_SET);
+            VehicleRoutingSolution solution = importer.readSolution(dataSetFile);
+            vehicleRoutingSolutions[i] = solution;
+        }
+
+        solverFactory = SolverFactory.createFromXmlResource(vehicleRoutingApp.getSolverConfig());
+        SolverConfig solverConfig = solverFactory.getSolverConfig();
+
+        solverConfig.getPhaseConfigList().forEach(phaseConfig -> {
+            if (LocalSearchPhaseConfig.class.isAssignableFrom(phaseConfig.getClass())) {
+                phaseConfig.setTerminationConfig(new TerminationConfig().withStepCountLimit(STEP_LIMIT));
+            }
+        });
+        solverConfig.setMoveThreadCount(MOVE_THREAD_COUNT);
+        solverConfig.setEnvironmentMode(EnvironmentMode.REPRODUCIBLE);
+    }
+
+    @Test
+    public void multiThreadedSolvingIsReproducible() {
+        IntStream.range(0, REPETITION_COUNT).forEach(iteration -> solveAndCompareWithPrevious(iteration));
+    }
+
+    private void solveAndCompareWithPrevious(final int iteration) {
+        Solver<VehicleRoutingSolution> solver = solverFactory.buildSolver();
+        VehicleRoutingSolution bestSolution = solver.solve(vehicleRoutingSolutions[iteration]);
+        vehicleRoutingSolutions[iteration] = bestSolution;
+
+        if (iteration > 0) {
+            VehicleRoutingSolution previousBestSolution = vehicleRoutingSolutions[iteration - 1];
+            assertEquals(bestSolution.getScore(), previousBestSolution.getScore());
+        }
+    }
+
+}
```

Comment:
@rsynek Thank you for keeping the limit of 128 chars per line. It makes the review on GitHub comfortable.